pyproject.toml

```diff
@@ -37,22 +37,31 @@ polars = ["polars>=0.20.3"]
 dask = ["dask[dataframe]>=2024.8"]
 duckdb = ["duckdb>=1.0"]
 ibis = ["ibis-framework>=6.0.0", "rich", "packaging", "pyarrow_hotfix"]
-dev = [
+tests = [
   "covdefaults",
-  "pre-commit",
   "pytest",
   "pytest-cov",
   "pytest-randomly",
   "pytest-env",
   "hypothesis",
   "typing_extensions",
 ]
+typing = [
+  "typing_extensions",
+  "mypy~=1.15.0",
+]
+dev = [
+  "pre-commit",
+  "narwhals[tests]",
+  "narwhals[typing]",
```

Comment:
I think we should separate the dev dependencies into `tests` and `typing`.

The latest pinned `mypy` is available for python >=3.9 and if we simply include it in `dev` our python 3.8 tests will fail.